wordpress-mobile/WordPress-iOS
# Comments: Remove unified list feature flag

This PR removes the `unifiedCommentsAndNotificationsList` feature flag – making the unified list permanent. This also removes some classes:

- `CommentsTableViewCell` – used in Site Comments before the unified list implementation.
- `NoteTableHeaderView`, `NoteTableViewCell`, and `NoteUndoOverlayView` – these are solely used in Notifications list. After the unified list, they are replaced with `ListTableViewCell` (and `ListSimpleOverlayView` to replace the undo overlay functionality).

In addition, to improve the visual look, a 10pt padding was [added for the first header view](https://github.com/wordpress-mobile/WordPress-iOS/pull/16885#issuecomment-883824926) in Notifications. Previously, I added this via code since we needed to support both new and old implementation. With the old implementation removed, this padding is now moved to the Storyboard instead to reduce clutter.

## To test

- Verify that the comments list is displayed correctly in My Site > Comments.
- Verify that the Notifications list is displayed correctly.

## Regression Notes
1. Potential unintended areas of impact
Should be none.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manually tested to ensure there is no impact on the current state.

3. What automated tests I added (or what prevented me from doing so)
n/a.

PR submission checklist:
